app/Http/Controllers/Common/SettingsController.php

```diff
@@ -554,4 +588,154 @@ public function postdebugSettings(Request $request)
 
         return redirect()->back()->with('success', \Lang::get('message.updated-successfully'));
     }
+
+    public function settingsPayment(Setting $settings, Request $request)
+    {
+        try {
+            $from = $request->input('from');
+            $till = $request->input('till');
+
+            return view('themes.default1.common.payment-log', compact('from', 'till'));
+        } catch (\Exception $ex) {
+            return redirect()->back()->with('fails', $ex->getMessage());
```

Comment:
From now on we log exceptions and display proper English error messages. Check this everywhere